crates/iceberg/src/table.rs

```diff
@@ -203,7 +203,7 @@ impl Table {
 
     /// Creates a metadata table which provides table-like APIs for inspecting metadata.
     /// See [`MetadataTable`] for more details.
-    pub fn metadata_table(self) -> MetadataTable {
+    pub fn metadata_table(&self) -> MetadataTable<'_> {
```

Comment:
Not strictly related to this PR. Just updating this to be a reference because the ownership isn't necessary